### PR TITLE
275 update procurement unit map

### PR DESCRIPTION
**Pull request by danieldunderfelt**
Closes https://github.com/HSLdevcom/bultti/issues/275

UI PR: https://github.com/HSLdevcom/bultti-ui/pull/88 (only minor schema updates)

- Clear up procurement unit fetching and add some comments
- Update contract -> procurement unit map creation
- Update ContractManager to fetch contracts by date and from linked pre-inspections
- Update contract manager method calls to use date

### CODE REVIEW

File: src/contract/ContractManager.ts [hunk 1]
```diff
@@ -3,132 +3,90 @@ import { getRepository } from 'typeorm'
...
 import { ProcurementUnitContractMap } from '../inspection/inspectionTypes'
 
+/**
+ * The contract manager provides contract data. Contract selection is based on pre-inspections.
+ * If the manager is provided with a PRE inspection, the logic is simple, as it just uses
+ * the inspectionStartDate.
+ *
+ * If a POST inspection is provided, each contract value must be requested with a date
+ * that is used in contract selection. The linked pre-inspection for that date is
+ * used to select the correct contract to use.
```
[the remark this answers is not in the document]
Comment (Divinare) [reply src/contract/ContractManager.ts:21]: Also I was thinking of making the date optional and in the start of the functions, check if given preInspection + date, you could throw an error (perhaps I'm overthinking)

File: src/contract/ContractManager.ts [hunk 2]
```diff
@@ -3,132 +3,90 @@ import { getRepository } from 'typeorm'
...
+    procurementUnitId: string,
+    date: string = this.inspection.inspectionStartDate
+  ): Promise<Contract> {
     let realId: string = procurementUnitId
 
-    if (procurementUnitId.length < 20) {
-      let foundId = await this.findRealProcurementUnitId(procurementUnitId)
+    // If the provided ID is shorter than UUID length, we got a "kohdetunnus"
+    // and need to find out the UUID for the unit in Bultti's database.
+    if (procurementUnitId.length < 36) {
```
Comment (Divinare) [inline src/contract/ContractManager.ts:60]: Not really a fan of supporting both .id and .procurementUnitId if not have to

**Review by Divinare — CHANGES_REQUESTED** [review]
ContractManager makes now sense!

Couldn't test though, not sure if it's just for me but it doesn't let me create preInspection (yes, I pulled have the latest changes :P)

create_preInspection

File: src/contract/ContractManager.ts (hunk 2 above)
Comment (danieldunderfelt) [reply src/contract/ContractManager.ts:60]: Yeah, well, departures from JORE only have the JORE ID (kohtunnus) and it has to be mapped to Bultti Procurement Unit UUID at some point. This is a logical place to do so and it is nicely invisible to the caller of ContractManager methods.

File: src/contract/ContractManager.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (danieldunderfelt) [reply src/contract/ContractManager.ts:21]: Good suggestion! I can expand on the comment further.

**Pull request merged**